workers/unity/Packages/com.improbable.gdk.core/GameObjectRepresentation/IReaderWriterCreator.cs

```diff
@@ -0,0 +1,9 @@
+using Unity.Entities;
+
+namespace Improbable.Gdk.Core.MonoBehaviours
+{
+    internal interface IReaderWriterCreator
+    {
+        object CreateReaderWriter(Entity entity, EntityManager entityManager, ILogDispatcher logDispatcher);
```

Comment:
Can the return type be more specific? `IReader` ?

Reply:
The problem here is that `IReader` requires a generic parameter, e.g. `IReader<SpatialOSPosition>`. Meanwhile `IReaderInternal` is an internal interface and C# says an interface may only be implemented using public methods. :-/
Thoughts? @firtina-improbable

Reply:
In which case you could either get IReader to inherit something not generic or you just get rid of it. 
Given that we only seem to explicitly implement it, which means it mostly can't be used as an interface, and you need to know the type information anyway to cast it I don't think it needs to exist.

Reply:
Discussed offline, if the impl class is internal it's safe to return also-internal types

Reply:
The Create methods now return `IReaderInternals`